samza-core/src/main/java/org/apache/samza/storage/StoreBackupEngine.java

```diff
@@ -0,0 +1,185 @@
+    LOGGER.info("Total size deleted: " + totalBytesDeleted);
+
+    return true;
+  }
+
+
+  // Method to find new files in backup compared to local
+  public ArrayList getNewFilesInBackupToLocal(Path localPath) throws IOException {
+    HashMap<String, File> localFileHmap = new HashMap<String, File>();
+    ArrayList<FileStatus> filesNotInLocalList = new ArrayList<FileStatus>();
+    // Store all the Rocksdb files in hashmap for quick comparison
+    File dbDir = new File(localPath.toString());
+    File[] filesList = dbDir.listFiles();
+    for (File dbFile : filesList) {
+      String fileName = dbFile.getName();
+      localFileHmap.put(fileName, dbFile);
+    }
+
+    // Get all files in the hdfs backup and compare to local. Add if different/doesn't exist
+    FileStatus[] backupFileIterator = fs.listStatus(hdfsPath);
+    for (FileStatus backupFile : backupFileIterator) {
+      String fileName = backupFile.getPath().getName();
+
+      // If found locally, check filesize to see if needed
+      if (localFileHmap.containsKey(fileName)) {
+        // TODO: This may not be good enough check if things fail to copy or weird things happen, probably an edge case here to consider
+        if (backupFile.getLen() > localFileHmap.get(fileName).length()) {
+          filesNotInLocalList.add(backupFile);
+        }
+      // If not found, automatically add to list
+      } else {
+        filesNotInLocalList.add(backupFile);
+      }
+    }
+
+    return filesNotInLocalList;
+  }
+
+  // Method to find new/Modified in local directory compared to backup
+  public ArrayList<File> getNewFilesInLocalToBackup(Path localPath) throws IOException {
+    HashMap<String, FileStatus> hdfsFileHmap = new HashMap<String, FileStatus>();
+    ArrayList<File> filesNotInHDFSList = new ArrayList<File>();
+    // Store all the backup files in hashmap for quick comparison
+    FileStatus[] backupFileIterator = fs.listStatus(hdfsPath);
+    for (FileStatus file : backupFileIterator) {
+      String fileName = file.getPath().getName();
+      hdfsFileHmap.put(fileName, file);
+    }
+
+    // Get all files in the local directory and compare to backup. Add if different/doesn't exist
+    File localDir = new File(localPath.toString());
+    File[] filesList = localDir.listFiles();
+    for (File dbFile : filesList) {
+      String fileName = dbFile.getName();
+
+      if (hdfsFileHmap.containsKey(fileName)) {
+        // TODO: This may not be good enough check if things fail to copy or weird things happen, probably an edge case here to consider
+        // If found locally, check filesize to see if needed
+        if (dbFile.length() > hdfsFileHmap.get(fileName).getLen()) {
+          filesNotInHDFSList.add(dbFile);
+        }
+      } else {
+        filesNotInHDFSList.add(dbFile);
+      }
+    }
+
+    return filesNotInHDFSList;
+  }
+
+  // Method to close hdfs file system
+  public void closeFileSystem() throws IOException {
```

Comment:
If things need to be started/stopped in this class, this class should have start/stop methods itself. Let's not expose implementation detail in its interface (e.g., the fact that this has a 'filesystem').